extension/scripts/features/racing-filter/ttRacingFilter.js

```diff
@@ -0,0 +1,352 @@
+				const feeAmount = parseInt(feeText, 10);
+				hasFee = feeAmount > 0;
+			}
+
+			// Drivers
+			const driversElement = li.querySelector("li.drivers");
+
+			let isFull = false;
+			let maxDriversAllowed = 0;
+			if (driversElement) {
+				// Extract only numbers and slashes from the text
+				const text = driversElement.textContent.replace(/\s+/g, ""); // remove whitespace
+				// Match the pattern "number / number"
+				const match = text.match(/(\d+)\/(\d+)/);
+
+				if (match) {
+					let driversJoined = 0;
+					driversJoined = parseInt(match[1], 10);
+					maxDriversAllowed = parseInt(match[2], 10);
+					if (driversJoined >= maxDriversAllowed) {
+						
+						isFull = true;
+					}
+				}
+			}
+
+			// Laps
+			const laps = parseInt(li.querySelector(".laps").textContent.match(/\d+/)[0], 10);
+
+			// Start time
+			const timeText = li.find(".event-wrap .startTime").textContent; // Format can be : "waiting", "26 m", "10 h 30 m"
+
+			let totalHours = 0;
+
+			if (!timeText || timeText.toLowerCase() === "waiting") {
+				totalHours = 0;
+			}
+			else {
+				// normalize text
+				const cleanText = timeText.trim().toLowerCase();
+
+				// extract hours and minutes explicitly
+				const hoursMatch = cleanText.match(/(\d+)\s*h/);
+				const minsMatch  = cleanText.match(/(\d+)\s*m/);
+
+				const hours = hoursMatch ? parseInt(hoursMatch[1], 10) : 0;
+				const minutes = minsMatch ? parseInt(minsMatch[1], 10) : 0;
+
+				// convert to whole hours
+				totalHours = hours + Math.floor(minutes / 60);
+			}
+
+			// Track name
+			const trackElement = li.querySelector("li.track");
+			const trackName = Array.from(trackElement.childNodes)
+				.filter(node => node.nodeType === Node.TEXT_NODE)
+				.map(node => node.textContent.trim())
+				.join(" ")
+				.trim();
+
+			// Race name
+			const raceName = li.find(".event-wrap .name").textContent;
+
+			/*************************************************/
+			/**    Apply filters and update race list       **/
+			/*************************************************/
+            // Hide races
+			if (hideRacesFilter.length && hideRacesFilter.includes("protected") && isProtected) {
+				hideRow(li);
+				continue;
+			}
+            if (hideRacesFilter.length && hideRacesFilter.includes("incompatible") && isIncompatible) {
+				hideRow(li);
+				continue;
+			}
+            if (hideRacesFilter.length && hideRacesFilter.includes("paid") && hasFee) {
+				hideRow(li);
+				continue;
+			}
+            if (hideRacesFilter.length && hideRacesFilter.includes("full") && isFull) {
+				hideRow(li);
+				continue;
+			}
+
+			// Max Drivers
+			if ((maxDriversAllowed < driversMin) || (maxDriversAllowed > driversMax)) {
+				hideRow(li);
+				continue;
+			}
+
+			// Max Laps
+			if ((laps < minLaps) || (laps > maxLaps)) {
+				hideRow(li);
+				continue;
+			}
+
+			// Start time
+			if ((timeStart && totalHours < timeStart) || (timeEnd !== 48 && totalHours >= timeEnd)) {
```

Comment:
This also hides 'waiting' races if I set the filter to 0-0, which is not what I would expect.